homeassistant/components/matter/light.py

```diff
@@ -302,12 +282,13 @@ def _update_from_device(self) -> None:
         """Update from device."""
         if self._attr_supported_color_modes is None:
```

Comment:
I'd consider moving what's under this if-statement to its own method, to be called from `async_added_to_hass`, before the call to `self._update_from_device()`